### DISCUSSION TITLE
Use lathe with power servo motor as spindle drive

### DISCUSSION THREAD

**Discussion by digidiver11** [grblHAL/core#365] [General]
Hello,

is there a possibility with GrblHAL to control a lathe - spindle with servo motor ??
The servo controller needs input pulses up to 500kHz.
Connections via analog input or RS485 are also possible.

By a servo motor an indexing can be used easily, for milling work on the lathe.
Also the spindle synchronization for thread cutting should be easier possible, or ?

**Comment by terjeio** [comment]
Analog input supported via PWM and RS485 control is possible but driver code has to be written?
Pulse control is via step/dir?

Spindle syncronization is not made easier since code for controlling the servo must be added. Depending on the motor interface this might be a more or less of a challenge since coordinated moves is needed - and the servo motor needs to be on an infinite length axis, which is not yet supported.

**Comment by digidiver11** [comment]
Hello Terje Io,

Thank you for your quick reply. I think the implementation is not so difficult.

1. PWM: The servo controller is controlled analog with -10 to +10V control voltage.
	By switching from + to - the direction of rotation is changed.
	I.e. the analog output is simply inverted or not.
	I am designing the circuit for this.
	I will gladly provide you with this circuit.

2. RS485: In my opinion you can think about this later.

3. the pulse control is done by Step/Dir?    YES !

4. spindle synchronization:
   I think this is not so complicated. The servo controller provides a complete
   encoder signal, i.e. A,B and X. Even as symmetrical signal.
   However, this means that the ratio to the spindle is predefined, 1:1.
   Per revolution the controller supplies 2500 umpulses.

   I.e. the spindle speed can be easily realized by the PWM control.
   Then the synchronization as it is intended in your software should be easily realizable.
   Here there is only the hurdle with the high number of pulses per revolution.

What I personally find very interesting is that additionally the spindle is controlled by Step/Dir as axis A.

One more question, do you have boards from your "CNC_Breakout_Nucleo64", from which you could send me a PCB?
you could send me. Of course I would take over the costs for it.
I like your hardware very much.

Best regards

**Reply by terjeio** [reply]
1. Combined with the encoder signal: this can use the current spindle sync code.
Publishing the circuit might be of interest to others, but I do not need it myself - at least for now.
3. Ok.
4. This is the complicated one, if the servo is driven as an axis (A) then this axis has to be RPM controlled (independently of the other axes) and, at least in principle, support infinite rotation. If not commands for spindle synced motion (G33 and G76) will not work. If instead you generate gcode that uses the spindle as a regular rotational axis then it should work - so easy for sure (at least for me).

> One more question, do you have boards from your "CNC_Breakout_Nucleo64", from which you could send me a PCB?

Bare boards are still available, [see here](https://github.com/terjeio/CNC_Breakout_Nucleo64).
